File: Curso.java
class Curso {
    private String nombreCurso;
    private Horario horario; // Composición

    public Curso(String nombreCurso, String dias, String horas) {
        this.nombreCurso = nombreCurso;
        // El horario nace y muere con el curso
        this.horario = new Horario(dias, horas);
    }

    public String getNombreCurso() { return nombreCurso; }
    public void setNombreCurso(String nombreCurso) { this.nombreCurso = nombreCurso; }
    public Horario getHorario() { return horario; }
    public void setHorario(Horario horario) { this.horario = horario; }

    @Override
    public String toString() {
        return "Curso: " + nombreCurso + " | Horario: " + horario.toString();
    }
}

File: Python/curso.py
from horario import Horario


class Curso:
    def __init__(self, nombre_curso: str, dias: str, horas: str):
        self.nombre_curso = nombre_curso
        # El horario nace y muere con el curso (composición)
        self.horario = Horario(dias, horas)

    def get_nombre_curso(self) -> str:
        return self.nombre_curso

    def set_nombre_curso(self, nombre_curso: str):
        self.nombre_curso = nombre_curso

    def get_horario(self) -> Horario:
        return self.horario

    def set_horario(self, horario: Horario):
        self.horario = horario

    def __str__(self):
        return f"Curso: {self.nombre_curso} | Horario: {self.horario}"
